File: SuperFood/Typescript/HomeDashboard/Models/Product.ts
/// <reference path="../../../scripts/typings/angularjs/angular.d.ts" />
/// <reference path="../../../scripts/typings/angularjs/angular-route.d.ts" />

module SuperFood.HomeDashboardApp.Models {

    export class Product {
        Id: number;
        Name: string;
        Description: string;
        Price: number;
        InStock: boolean;
        Details: Array<string>;
        ProductType: SuperFood.AdministrationApp.Models.ProductType;
        IsDeleted: boolean;
        Amount: number;
    }
}

File: SuperFood.Shared/Data/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity.Spatial;

namespace SuperFood.Shared.Data.Models
{

    [Table("Product")]
    public partial class Product
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public Product()
        {
            OrderProducts = new HashSet<OrderProduct>();
            Categories = new HashSet<Category>();
        }

        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        public string Name { get; set; }

        public string Details { get; set; }

        public string Description { get; set; }

        public decimal Price { get; set; }

        public bool InStock { get; set; }

        public int? ProductTypeId { get; set; }

        public bool IsDeleted { get; set; }

        [StringLength(255)]
        public string Topings { get; set; }

        [StringLength(50)]
        public string Image { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<OrderProduct> OrderProducts { get; set; }

        public virtual ProductType ProductType { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Category> Categories { get; set; }

    }
}
